nae-object-library/src/main/java/com/netgrif/application/engine/objects/auth/domain/Realm.java

```diff
@@ -4,60 +4,133 @@
 import jakarta.validation.constraints.NotNull;
 import lombok.Data;
 
-import java.time.Duration;
 import java.io.Serial;
 import java.io.Serializable;
+import java.time.Duration;
 import java.util.ArrayList;
 import java.util.HashSet;
 import java.util.List;
 import java.util.Set;
 
+/**
+ * Represents a security realm in the authentication system.
+ * A realm is a security context that contains authentication methods, user IDs, 
+ * and security settings for a group of users.
+ */
 @Data
 public class Realm implements Serializable {
 
+    /**
+     * Serial version UID for serialization support.
+     */
     @Serial
     private static final long serialVersionUID = -162168235241317688L;
 
+    /**
+     * The unique identifier of the realm.
+     */
     private String id;
 
+    /**
+     * Indicates whether this realm is the default realm in the system.
+     * Only one realm can be set as default.
+     */
     @NotNull
     private boolean defaultRealm = false;
 
+    /**
+     * The name of the realm. This field is required.
+     */
     @NotNull
     private String name;
 
+    /**
+     * Optional description of the realm's purpose or characteristics.
+     */
     private String description;
 
+    /**
+     * List of authentication methods configured for this realm.
+     * Marked as transient to prevent serialization.
+     */
     private transient List<AuthMethodConfig<?>> authMethods = new ArrayList<>();
 
+    /**
+     * Set of user IDs associated with this realm.
+     */
     private Set<String> userIds = new HashSet<>();
```

Comment:
I think having all users' IDs in the realm object is dangerous. If a realm has many users, this Set can get large and slow to read from the database. As a user has a reference to which realm belongs to, finding all users of a realm is one simple query away.